### PR TITLE
fix(components/ag-grid): improve schematic for AG Grid 29 changes

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2665124](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2665124)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1675/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1675/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1675/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1675/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1675/playground/)

File: libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.spec.ts [hunk 1]
```diff
@@ -1,37 +1,48 @@
...
-          architect: {
-            build,
-          },
-        },
-      },
+    let tree: Tree;
+    if (appOrLib === 'application') {
+      tree = await createTestApp(runner, { projectName: 'test' });
+    } else {
+      tree = await createTestLibrary(runner, { projectName: 'test' });
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.spec.ts:22]: ☝️ Love this `setupTest` function.

File: libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.ts [hunk 2]
```diff
@@ -35,25 +35,39 @@ function removeAgGridCssFiles(): Rule {
...
-          )
+          targetDefinition?.options &&
+          Array.isArray(targetDefinition.options['styles'])
         ) {
-          build.options['styles'].push(
-            'node_modules/@skyux/ag-grid/css/sky-ag-grid.css'
+          targetDefinition.options['styles'] = (
+            targetDefinition.options['styles'] as string[]
+          ).filter(
+            (cssFile) => !cssFile.startsWith('ag-grid-community/dist/styles/')
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.ts:56]: Wouldn't this remove `ag-grid-community` stylesheets before checking if SKY UX/AG Grid stylesheets are being used? There's a slim chance, but one of our external consumers might have a `build` architect that uses the `ag-grid-community` stylesheet but doesn't use SKY UX components. In that case, we'd want to leave the `styles` alone. Maybe put this logic in the `if` block below?

Comment (johnhwhite) [reply libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.ts:56]: Good call. Updated.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**